common/lib/types/errorinfo.js

```diff
@@ -1,10 +1,11 @@
 var ErrorInfo = (function() {
 
-	function ErrorInfo(message, code, statusCode, cause) {
+	function ErrorInfo(message, code, statusCode, cause, recoveryStrategy) {
```

Comment:
Should we have a category of error codes that trigger reattach? It's nice to propagate it in the way you're doing it here, but polluting `ErrorInfo` just in order to be able to do that is a problem I think.

The other way to approach this possibly is that the thing that gets thrown by `decode()` doesn't have to be the `ErrorInfo` - it can be some other `Error` type that wraps an `ErrorInfo`, with the other `retryStrategy` information  implicitly or explicitly embedded in that type.

Reply:
a category seems like overkill -- we get along fine currently with only one error code for all types of decode errors (40013), adding a second one for decode errors that trigger a reattach seems fine to me

Reply:
Fixed in https://github.com/ably/ably-js/pull/615/commits/2ff1bc3306b914d70079848ba37fbd51f84deea1